### PR TITLE
feat(widget): widgets emit widget.effects

### PR DESCRIPTION
**Pull request by EdAyers**
See https://github.com/leanprover-community/lean/pull/361

### CODE REVIEW

File: infoview/info.tsx [hunk 1]
```diff
@@ -75,7 +74,7 @@ export function Info(props: InfoProps) {
                         </div> }
                 </div>
                 <div>
-                    <Widget widget={widget} fileName={loc.file_name} onEdit={onEdit} />
+                    <Widget widget={widget} fileName={loc.file_name} />
```
Comment (gebner) [inline infoview/info.tsx:77]: You should merge master.  This was already changed there.

File: infoview/widget.tsx [hunk 2]
```diff
@@ -95,10 +119,17 @@ export function Widget({ widget, fileName, onEdit }: WidgetProps): JSX.Element {
...
         if (record.status === 'success' && record.widget) {
+            const effects: WidgetEffect[] | undefined = (record as any).effects;
+            if (effects) {
+                for (const effect of effects) {
+                    applyWidgetEffect(widget, fileName, effect);
+                }
+            }
             setHtml(record.widget.html);
         } else if (record.status === 'edit') {
+            // @deprecated case
```
Comment (gebner) [inline infoview/widget.tsx:130]:
```suggestion
            // Lean < 3.17
```

File: infoview/info.tsx (hunk 1 above)
Comment (EdAyers) [reply infoview/info.tsx:77]: eek sorry I thought I did merge master!

**Comment by gebner** [comment]
The code here looks reasonable enough.  This is still WIP because you want to PR the event types to lean-client-js, right?

**Pull request merged**